## 1 T-test

### 1.1 Conditions to use T-test

1. Population distribution is assumed to be Normal.
2. Sample distribution is continuous.
3. Sample drawn in random.
4. homogeneity of variance

When sample size is less than 30 and population standard deviation $(\sigma)$ is not know.

> **Note**:  
> For sample size greater than 30, T-test behaves like Z-test.

## 2 One Sample T-test

### 2.1 Nature of hypothesis

- $H_0: \mu = \mu_0$
- $H_a: \text{$\mu \neq \mu_0$ (or $\mu > \mu_0$, $\mu < \mu_0$)}$

### 2.2 Test Statistic

The name of the test statistic in T-test is called as T-statistic.

$
\begin{align}
\large
\text{T-Statistic} = \frac{x - \mu}{\frac{s}{\sqrt{n}}}
\end{align}
$

### 2.3 API

```python
from scipy.stats import ttest_1samp
```

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_1samp.html

## 3 Independent Two Sample T-test

### 3.1 Nature of hypothesis

- $H_0: \mu_1 = \mu_2$
- $H_a: \text{$\mu_1 \neq \mu_2$ (or $\mu_1 > \mu_2$, $\mu_1 < \mu_2$)}$

### 3.2 API

```python
from scipy.stats import ttest_ind
```

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html

## 4 Paired Two sample T-test

### 4.1 Nature of hypothesis

- $H_0: \mu_d = 0$
- $H_a: \text{$\mu_d \neq 0$ (or $\mu_d > 0$, $\mu_d < 0$)}$

### 4.2 API

```python
from scipy.stats import ttest_rel
```

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_rel.html

## 5 Quizzes

In [1]:
import numpy as np
from scipy.stats import ttest_1samp, ttest_ind, ttest_rel, norm

### Quiz #1

In [2]:
iq_scores = [110, 105, 98, 102, 99, 104, 115, 95]

#### Solution

In [3]:
# Given that average IQ of human beings are considered as 100.
mu = 100  # Population mean.
n = len(iq_scores)

# H_0: mu <= 100  Given IQ scores are closer to global average.
# H_a: mu > 100   Given IQ scores are larger than global average.

In [4]:
# Distribution: T distribution
# Test statistic: T-score
# Significance level: 0.05
alpha = 0.05

In [5]:
# Since H_a has > symbol, perform right tailed test.

In [6]:
mu_0 = np.mean(iq_scores)  # Sample mean
mu_0.item()

103.5

In [7]:
t_stat, p_value = ttest_1samp(a=iq_scores, popmean=mu, alternative="greater")
t_stat.round(6).item(), p_value.round(6).item()

(1.507157, 0.08775)

In [8]:
if p_value < alpha:
    print("Reject Null hypothesis")
else:
    print("Failed to reject Null hypothesis")

Failed to reject Null hypothesis


Average IQ of sample, 103, is **NOT** statistically significant, hence given IQ scores are closer to global average.

In [9]:
sample_se = np.std(iq_scores) / np.sqrt(n)
sample_se.round(6).item()

2.172268

In [10]:
t_critical = norm.ppf(1 - (alpha / 2))
t_critical.round(6).item()

1.959964

In [11]:
# x = mu + (t_score * sample_se)
x = mu + (t_critical * sample_se)
x.round(1).item()

104.3

Sample size and population standard deviation criteria for Z-test vs T-test:

<div style="display: inline-block">

| Sample size $n$ and Population Standard Deviation $\sigma$                      | Hypothesis Test |
| :------------------------------------------------------------------------------ | :-------------- |
| Sample size $n <  30$ and population standard deviation $\sigma$ is known.      | Z-test          |
| Sample size $n <  30$ and population standard deviation $\sigma$ is NOT known.  | T-test          |
| Sample size $n \ge 30$ and population standard deviation $\sigma$ is known.     | Z-test          |
| Sample size $n \ge 30$ and population standard deviation $\sigma$ is NOT known. | Z-test / T-test |

</div>